iup/__init__.py

```diff
@@ -5,9 +5,24 @@
 import abc
 from typing_extensions import Self
 from typing import List
+import re
 
 
-class UptakeData(pl.DataFrame, metaclass=abc.ABCMeta):
+class ValidateData(pl.DataFrame, metaclass=abc.ABCMeta):
```

Comment:
Do we really need separate classes for observed data and forecasts? In essence, they are both data frames with dates and uptake values. The only difference is in how we interpret them, not in their structure or contents. Maybe we can collapse some of this code...

Reply:
Maybe this will be useful when we produce the forecasts that do not have the same format as observed data? Like in sample forecast, the forecast will be a distribution instead of a single Series.

Reply:
In the case of a sample forecast, the distribution will be given by many (say 1000) samples drawn from it, right? So this is a data frame with a date column and 1000 uptake columns. It still seems to me like this structurally similar enough to be the same class as the observed uptake data, if we design our tests and validate methods right. But maybe I should think about this a little more.